### DISCUSSION TITLE
GitHub special Markdown blocks aren't rendered in README section

### DISCUSSION THREAD

**Discussion by kbdharun** [rust-lang/crates.io#8506] [Feature Requests]
### Current Behavior

Currently, crates display the blockquote as it is in Markdown.

![image](https://github.com/rust-lang/crates.io/assets/26346867/d235c86c-3389-4773-afbb-5a7f39acf83a)

### Expected Behavior

GitHub supports custom rendering for descriptions enclosed inside blockquote starting with [!NOTE], [!TIP], [!IMPORTANT], [!WARNING], [!CAUTION] and this is widely used by many projects.

Syntax:

```md
> [!NOTE]  
> Highlights information that users should take into account, even when skimming.

> [!TIP]
> Optional information to help a user be more successful.

> [!IMPORTANT]  
> Crucial information necessary for users to succeed.

> [!WARNING]  
> Critical content demanding immediate user attention due to potential risks.

> [!CAUTION]
> Negative potential consequences of an action.
```
Which looks like this once rendered:

![image](https://github.com/rust-lang/crates.io/assets/26346867/37d8117f-c635-4d64-bd1c-57a400077e80)

These descriptions in Crates README can be rendered similarly to how it is shown in GitHub.

Check out https://github.com/orgs/community/discussions/16925 for more information.

### Steps To Reproduce

1. Go to a Crate using this kind of syntax in the README file i.e. <https://crates.io/crates/tlrc>.
2. Navigate to the note or relevant section.

### Environment

- **Browser**: Google Chrome Version 124.0.6367.60 (Official Build) (64-bit)
- **OS**: Fedora Linux 39 (Workstation Edition)

**Comment by fwcd** [comment]
Was just about to open the same feature request, would love to see this being added!

Though it seems like the Markdown rendering is largely delegated to [`comrak`](https://github.com/kivikakk/comrak), so it might be worth filing the feature request there:

https://github.com/rust-lang/crates.io/blob/e01e384a54d065b7244254ac82b1712d32ba75cb/crates/crates_io_markdown/lib.rs#L62-L106

**Reply by Arteiii** [reply]
I have created an issue in the [kivikakk/comrak](https://github.com/kivikakk/comrak) repository proposing the addition of special GitHub Markdown blocks. You can view the issue [here](https://github.com/kivikakk/comrak/issues/518)

I believe that adding support for these blocks would greatly improve the readability and usability of the documentation on crates.io as well.

However, I am not sure if I have the skills to implement this feature myself. If anyone is interested in helping, or has any suggestions on how to proceed, your input would be greatly appreciated.

**Reply by kbdharun** [reply]
Not sure how I missed this reply in my notifications 😅 , thanks for the code block @fwcd. 

Support for this has been added upstream as mentioned by @Arteiii below, so IG the `lib.rs` file can be updated for the same after a new comrak release.

**Comment by Arteiii** [comment]
https://github.com/kivikakk/comrak/pull/519

merged

either wait for release or add as git dependency

**Reply by kbdharun** [reply]
Thanks for the update.

**Reply by Turbo87** [reply]
https://github.com/rust-lang/crates.io/pull/10428 updates us to the new comrak version that supports this. PRs for enabling that feature and styling it properly would be welcome 😉

**Reply by kbdharun** [reply]
> https://github.com/rust-lang/crates.io/pull/10428 updates us to the new comrak version that supports this. PRs for enabling that feature and styling it properly would be welcome 😉 

Great, will try enabling it and will open a PR after testing. (Done, https://github.com/rust-lang/crates.io/pull/10429)
